dvc/output/base.py

```diff
@@ -431,29 +452,49 @@ def move(self, out):
             self.repo.scm.ignore(self.fspath)
 
     def get_files_number(self, filter_info=None):
```

Comment:
get_files_number/get_dir_cache are needed mostly for push/pull and really shouldn't be in the Output, as those things should be figured out on the odb/object level.